core/src/test/scala/unit/kafka/zk/ZooKeeperTestHarness.scala

```diff
@@ -52,6 +56,22 @@ abstract class ZooKeeperTestHarness extends JUnitSuite with Logging {
     if (zookeeper != null)
       CoreUtils.swallow(zookeeper.shutdown())
     Configuration.setConfiguration(null)
+    assertNoBrokerControllersRunning()
```

Comment:
I haven't done a full review yet. Will do it in 30 mins or so. I'll also merge when I'm done. Thanks.